collectors/googlestackdriver/collector.js

```diff
@@ -0,0 +1,120 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2019, Alert Logic, Inc
+ * @doc
+ *
+ * googlestackdriver class.
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+'use strict';
+
+const moment = require('moment');
+const PawsCollector = require('@alertlogic/paws-collector').PawsCollector;
+const parse = require('@alertlogic/al-collector-js').Parse;
+const logging = require('@google-cloud/logging');
+
+
+const typeIdPaths = [
+    {path: ['protoPayload', 'type_url']}
+];
+
+class GooglestackdriverCollector extends PawsCollector {
+    constructor(context, creds) {
+        super(context, creds, 'googlestackdriver');
+    }
+    
+    pawsInitCollectionState(event, callback) {
+        // TODO: put in some more efficient catching up logic for historical logs. Stackdriver stores logs for 30 days
+        const startTs = process.env.paws_collection_start_ts ? 
+                process.env.paws_collection_start_ts :
+                    moment().toISOString();
+        const endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+        const initialState = {
+            since: startTs,
+            until: endTs,
+            poll_interval_sec: 1
+        };
+        return callback(null, initialState, 1);
+    }
+    
+    pawsGetLogs(state, callback) {
+        let collector = this;
+
+        // Start API client
+        const client = new logging.v2.LoggingServiceV2Client({
+            credentials: JSON.parse(this._pawsCreds.secret)
+        });
+
+
+        console.info(`GSTA000001 Collecting data from ${state.since} till ${state.until}`);
+
+        // TODO: figure out a better way to format this. I'm pretty sure that it needs the newlines in it.
+        const filter = `timestamp >= "${state.since}"
+timestamp < "${state.until}"`;
+        const formattedResourceNames = JSON.parse(process.env.google_project_ids).map(e => `projects/${e}`);
+
+        // TODO: check out how the "autopagination" functionality works on this.
+        client.listLogEntries({filter:filter, resourceNames: formattedResourceNames})
+            .then(responses => {
+                const resources = responses[0];
```

Comment:
why only first element instead of entire array?

Reply:
see example. 

https://googleapis.dev/nodejs/logging/latest/v2.LoggingServiceV2Client.html#listLogEntries-examples

It seems that the actual content is an array that is the first element of that array.